src/manifests_cephfs.py

```diff
@@ -174,13 +181,29 @@ def parameter_list(self) -> List[CephStorageClassParameters]:
         return sc_names
 
     def __call__(self) -> List[AnyResource]:
-        """Craft the storage class object."""
+        """Craft the storage class objects."""
+
         if cast(SafeManifest, self.manifests).purging:
             # If we are purging, we may not be able to create any storage classes
             # Just return a fake storage class to satisfy delete_manifests method
             # which will look up all storage classes installed by this app/manifest
             return [StorageClass.from_dict(dict(metadata={}, provisioner=self.PROVISIONER))]
-        return [self.create(class_param) for class_param in self.parameter_list()]
+
+        empty = []
+        if not self.manifests.config.get("enabled"):
+            # If cephfs is not enabled, we cannot add any storage classes
+            log.info("Skipping CephFS storage class creation since it's disabled")
+            return empty
+
+        try:
+            parameter_list = self.parameter_list()
+        except ValueError as err:
+            # If we cannot generate the parameter list, we cannot add
+            # any storage classes
+            log.error("Failed to list storage classes to add: %s", err)
+            return empty
```

Comment:
Can we just return the empty list instead of var?
```suggestion
        if not self.manifests.config.get("enabled"):
            # If cephfs is not enabled, we cannot add any storage classes
            log.info("Skipping CephFS storage class creation since it's disabled")
            return []

        try:
            parameter_list = self.parameter_list()
        except ValueError as err:
            # If we cannot generate the parameter list, we cannot add
            # any storage classes
            log.error("Failed to list storage classes to add: %s", err)
            return []
```

Reply:
we sure can